decidim-core/lib/decidim/dummy_translator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This Dummy translator recieves the field value
+  # and the locale of the field which has to be
+  # translated. It returns the appended value for both.
+  # This is for testing only.
+  class DummyTranslator
+    attr_reader :text, :original_locale, :target_locale, :resource, :field_name
+
+    def initialize(resource, field_name, text, original_locale, target_locale)
+      @resource = resource
+      @field_name = field_name
+      @text = text
+      @original_locale = original_locale
+      @target_locale = target_locale
+    end
+
+    def translate
+      translated_text = "#{target_locale} - #{text}"
+
+      # This is a Dummy Translator, it returns the translation
+      # instantly and is only used for testing.
+      # After integrating your translation service,
+      # you could create a job to perform the following
+      # storing operations.
+      if resource[field_name]["machine_translations"].present?
+        resource[field_name]["machine_translations"] = resource[field_name]["machine_translations"].merge(target_locale => translated_text)
+      else
+        resource[field_name] = resource[field_name].merge("machine_translations" => { target_locale => translated_text })
+      end
+
+      # rubocop:disable Rails/SkipsModelValidations
+      resource.update_column field_name.to_sym, resource[field_name]
```

Comment:
I'm not 100% comfortable with translation service directly updating a resource. Also, this would mean that for each translation service a developer might need to create a job for it, or do it inside the service.

What if we already create a job class that handles updating the resource so it can be reused in any translation service? This would solve also our problem with async translation services that send you back the translation via a webhook.

We could update the DummyTranslation to enqueue that job.

Something like:

```ruby
  def translate
    translated_text = "#{target_locale} - #{text}"

    MachineTranslationSaveJob.perform_later(resource, field_name, target_locale, translated_text)
  end

  module Decidim
    # This jobs is used by machine translation services to store the result of
    # a field translation. This way services don't need to care about how to
    # save it and also enables storing translations asynchronously when the
    # translation service returns the translated text in a webhook
    class MachineTranslationSaveJob < ApplicationJob
      queue_as :default

      def perform(resource, field_name, target_locale, translated_text)
        if resource[field_name]["machine_translations"].present?
          resource[field_name]["machine_translations"] = resource[field_name]["machine_translations"].merge(target_locale => translated_text)
        else
          resource[field_name] = resource[field_name].merge("machine_translations" => { target_locale => translated_text })
        end

        # rubocop:disable Rails/SkipsModelValidations
        resource.update_column field_name.to_sym, resource[field_name]
        # rubocop:enable Rails/SkipsModelValidations
      end
    end
  end

```

Reply:
We considered the option but discarded it in order to simplify the example code, but we can work on it!

Reply:
Done!